Framework/API/src/AlgorithmHistory.cpp

```diff
@@ -174,6 +174,23 @@ AlgorithmHistory_sptr AlgorithmHistory::operator[](const size_t index) const {
 }
 
 /**
+* Gets the value of a specified algorithm property
+* @param name ::  The property to find
+* @returns The string value of the property
+* @throw Exception::NotFoundError if the named property is unknown
+*/
+const std::string &
+AlgorithmHistory::getPropertyValue(const std::string &name) const {
+  for (auto &hist : m_properties) {
+    if (hist->name() == name) {
+      return hist->value();
+    }
+  }
```

Comment:
We have a slight disagreement on coding implementation, but it is not in violation of the coding standards, negative on performance, and 50/50 on readability.

If there isn't anything else could you pass this one please as it has been sat in limbo for almost 2 weeks.